sdk/core/System.ClientModel/src/Convenience/ClientResultException.cs

```diff
@@ -67,7 +67,7 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
 
     private static string CreateMessage(PipelineResponse response)
     {
-        response.BufferContent();
+        response.ReadContent();
```

Comment:
do we have an async variant of CreateMessage to uses async ReadContentAsync?

Reply:
This is called from the exception constructor, so it does not.

Reply:
Should we have an async factory instead using this ctor?

Reply:
Let me see what this looks like with the Azure.Core integration

Reply:
Yeah, I think it's good.  We have some shared source files that can't move to it immediately in the Core integration PR, but I was able to port pipeline instances to this factory method.  I'll update this PR now with the ClientModel side of the change and LMK if the API looks like something we're happy with